graylog2-web-interface/src/components/graylog/OverlayTrigger.tsx

```diff
@@ -55,15 +71,40 @@ class OverlayTrigger extends React.Component<Props, State> {
   }
 
   render() {
-    const { children, container, placement, overlay, rootClose, ...overlayProps } = this.props;
+    const { children, container, placement, overlay, rootClose, trigger, ...overlayProps } = this.props;
     const { show } = this.state;
 
-    const toggleShow = () => this.setState({ show: !show });
+    const setShow = (isShown: boolean) => this.setState({ show: isShown });
+    const toggleShow = () => setShow(!show);
+
+    const _onTrigger = {
+      click: {
+        onClick: toggleShow,
+      },
+      hover: {
+        onMouseEnter: () => setShow(true),
+        onMouseLeave: () => setShow(false),
+      },
+      focus: {
+        onFocus: () => setShow(true),
+        onBlur: () => setShow(false),
+      },
+    };
+
+    let triggerables = _onTrigger[trigger];
```

Comment:
I was also checking the PR, since I also pinged you yesterday about it 🙂 
It all looks good to me except for a Typescript error here: `TS2538: Type 'Triggers[]' cannot be used as an index type`. So I suggest making this assignment in a conditional, e.g.:

```
let triggerables;

if (Array.isArray(trigger)) {
  ...
} else {
  triggerables = _onTrigger[trigger];
}
```